app/src/main/java/com/craxiom/networksurveyplus/messages/QcdmWcdmaParser.java

```diff
@@ -126,6 +137,94 @@ private QcdmWcdmaParser()
                 isUplink, 0, 0, qcdmMessage.getSimId(), location);
     }
 
+    public static WcdmaRrc convertWcdmaRrcOtaMessage(QcdmMessage qcdmMessage, Location location, String deviceId, String missionId, String mqttClientId)
+    {
+        Timber.e("Handling an WCDMA RRC message MQTT");
+
+        final byte[] logPayload = qcdmMessage.getLogPayload();
+
+        final int channelType = logPayload[0] & 0xFF;
+        //final int rbid = logPayload[1] & 0xFF;
+        //final int messageLength = ParserUtils.getShort(logPayload, 2, ByteOrder.LITTLE_ENDIAN);
+
+        int headerLength;
+        int uarfcn = -1;
+        int psc = -1;
+
+        int subtype = getGsmtapWcdmaRrcChannelType(channelType);
+        if (subtype != -1)
```

Comment:
This if/else sequence can be moved to a stand alone method so that the method can be reused in both the WCDMA parsing methods. This will make it easier to update the code in the future so that we don't have to remember to make the change in both locations. If you are having a problem with doing that let me know and I can work something up for you to see how it can be done.